### PR TITLE
Use latest phpstan, drop composer-bin-plugin

### PR DESCRIPTION
**Pull request by staabm**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -37,7 +37,7 @@
         "ext-iconv": "Can be used as fallback when ext-mbstring is not available"
     },
     "require-dev": {
-        "bamarni/composer-bin-plugin": "^1.4",
```
Comment (staabm) [inline composer.json:40]: newer phpstan versions no longer define external dependencies. we don't need the composer bin plugin anymore

**Comment by staabm** [comment]
failling test looks unrelated

File: composer.json (hunk 1 above)
Comment (stof) [reply composer.json:40]: The main issue is that the 1.x branch supports PHP versions that are not supported by phpstan.

Comment (staabm) [reply composer.json:40]: ohh wow.. PHP5 is kind of old, didn't realize. thanks for the hint.

is it fine for the master branch?

Comment (stof) [reply composer.json:40]: For now, I'd rather keep the same setup in both branches. It makes it simpler to merge up.

Once the 2.0 rewrite is over and I can stop maintaining 1.x, I will reconsider

**Comment by stof** [comment]
I'm closing this PR:

- the setup with the separate install will be kept for now, as discussed in comments
- as long as we still have the huge baseline for the 1.x implementation, I'd rather keep a fixed version of phpstan so that CI builds don't randomly fail for contributors when phpstan gets updated and some reported errors change. This will also change once we reach the point where we drop the legacy code.

**Pull request closed without merging**